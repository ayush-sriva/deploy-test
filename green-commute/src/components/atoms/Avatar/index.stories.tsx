import { ComponentStory, ComponentMeta } from '@storybook/react';
import Avatar from '.';
import url from '../../../../public/assets/illustrations/user-img.png';

export default {
  title: 'components/atoms/Avatar',
  component: Avatar,
} as ComponentMeta<typeof Avatar>;

const Template: ComponentStory<typeof Avatar> = (args) => <Avatar {...args} />;

export const UserAvatar = Template.bind({});

UserAvatar.args = {
  alt: 'avatar img',
  src: `${url}`,
};
