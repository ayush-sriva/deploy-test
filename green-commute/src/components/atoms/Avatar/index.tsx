/**
 * @author Abhishek Bhemisetty <abhishek.bhemisetty@zemosolabs.com>
 */

import React from 'react';
import Avatar from '@mui/material/Avatar';
import { CommonProps } from '@mui/material/OverridableComponent';

interface AvatarProps {
  alt: string | undefined;
  src: string;
}

const AvatarComponent: React.FC<AvatarProps & CommonProps> = (props) => {
  const { alt, src } = props;

  return <Avatar alt={alt} src={src} />;
};

export default AvatarComponent;
