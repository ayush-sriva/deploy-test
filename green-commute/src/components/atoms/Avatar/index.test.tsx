import renderer from 'react-test-renderer';
import Avatar from '.';

describe('Avatar', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Avatar alt='test-user-avatar' src='../../../../public/assets/illustrations/user-img.png' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
