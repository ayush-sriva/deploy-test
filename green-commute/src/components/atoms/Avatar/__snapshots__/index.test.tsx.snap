// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
>
  <img
    alt="test-user-avatar"
    className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
    src="../../../../public/assets/illustrations/user-img.png"
  />
</div>
`;
